locksmith/src/controllers/v2/metadataController.ts

```diff
@@ -390,15 +391,40 @@ export class MetadataController {
     try {
       const lockAddress = Normalizer.ethereumAddress(request.params.lockAddress)
       const network = Number(request.params.network)
+      const keys = request.body
+
+      if (!keys?.keys) {
+        return response
+          .send({
+            message: 'Parameter `keys` is not present',
+          })
+          .status(500)
+      }
 
-      const results = await KeyMetadata.findAll({
-        where: {
-          address: lockAddress,
-          chain: network,
-        },
+      const mergeKeysMetadata = keys.map(async (key: any) => {
+        const mergedData = key?.keys?.map(async ({ owner, keyId }: any) => {
+          const keyOwner: string = owner.address
+          const keyHolderMetadata =
+            await lockOperations.getKeyHolderMetadataByAddress(
```

Comment:
I am not a huge fan of doing an SQL request for _each_, but I think that's a start.